### ISSUE TITLE
Landing page greeting uses preferred name when present

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-team#72972]
### User story

As an authenticated user of any type, I want the "Welcome" greeting on the MHV landing page to use the preferred name I entered into my profile and fall back to first name, so that the VA addresses me consistently by the name I prefer.

### Notes

- This is a follow-on to #68424, which is currently merged behind a feature flag. During the course of that ticket we learned about the availability of the preferred name field in the user's profile and the value of using it when available.
- When this ticket is complete, we can turn the greeting feature on in production. 
- Roll out to a % of users initially?
- Capitalization could be a concern – API seems to be sending names in all-caps. [These docs](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/products/identity-personalization/nametag/backend_documentation.md) might explain how to work with these fields (TravisH from Profile shared)

## Acceptance criteria

- The landing page greets the user by the preferred name from their profile when it's available, following typical capitalization practices on VAgov (i.e., capitalize first letter) 
- When preferred name isn't present, the page falls back to first name and then to no name (just "Welcome")
- Recommend a path to preserving capitalization of user's preferred name
- Implement behind a feature flag, will release to 25% of users when ready

## Tasks

- [x] Check in with the Auth Exp team (Tom Harrison crafted the code that fetches preferred name.) I think they were planning to use preferred name in the "auth menu" (I think it's called) in the top-right of header.
- [x] Confirm that preferred name will come back from the API in _all caps_
- [ ] Determine UX rules around capitalization – are names always transformed on the FE to first-letter-capitalized format?
- [x] QA: create test cases in TestRail
- [x] QA: confirm deployment to stage
- [x] QA: test feature/report bugs
- [ ] ~~QA: Regression test (if bugs found/fixed)~~
- [ ] Turned flag on in production to 25% of users

**Comment by radavis** [comment]
vets-api, User model, full_name_normalized method: https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/user.rb#L124

**Comment by wesrowe** [comment]
Carlos found this somewhere:
`const userFirstName = useSelector(state =>
    _.upperFirst(_.toLower(state.user.profile.userFullName.first)),
  );`

**Comment by dcloud** [comment]
Perhaps also relevant FE code: https://github.com/department-of-veterans-affairs/vets-website/blob/07bca52f81f19725ebcc64c7caf3b03d78573093/src/applications/personalization/profile/actions/personalInformation.js#L98-L107

May be where FE fetches "personal information" for Profile

**Comment by radavis** [comment]
Discussion on MPI validations/restrictions for the preferred name field (spaces are disallowed): https://dsva.slack.com/archives/C909ZG2BB/p1701882991635619

Asked Tom H. about uppercase name fields. He stated that preferred name is always returned as uppercase from the source system. This is pretty good evidence that name fields are uppercase somewhere along the line before saving.

Asked the same question in #vsp-identity. https://dsva.slack.com/archives/CSFV4QTKN/p1705676881137859

**Comment by radavis** [comment]
Per zoom w/ Wes R., wait until fetch request for personal information has completed before rendering `Welcome` component.

**Comment by wesrowe** [comment]
follow-up ticket written for moving the fetch of preferred name to root state, so that there is no flash or content shift on the MHV landing page. #74398

**Comment by wesrowe** [comment]
I just tested the new version and it looked great in Staging – the Welcome message appeared when the preferred name came back. no content flash and no shifting. well done. this can close as far as I'm concerned.

**Comment by fbonds** [comment]
Test cases created https://dsvavsp.testrail.io/index.php?/suites/view/1883&group_by=cases:section_id&group_order=asc&display_deleted_cases=0

I can't seem to get to the profile page to change the preferred name @wesrowe (see below).  

<img width="705" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN29XQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c810170b784a6f07af71f0cc5b0280230bdf2109/image.png" alt="image.png" />

**Comment by fbonds** [comment]
Successfully tested in Stage with vets.gov.user+1.  QA complete on this ticket. @wesrowe

**Comment by radavis** [comment]
Blocked by #74398

**Comment by wesrowe** [comment]
Update: this ticket was basically complete, but we couldn't release it because of feedback from stakeholders at the Weekly Design meeting. SHs felt that the salutation on the MHV landing page needs to agree with the name displayed in the auth menu in the header.

Updating the header led to API hits that were unwanted by the VA Profile back-end team, and a couple stories to address that (caching, etc).

Question:
@radavis, is the current state of the code for this ticket going to be re-used when we get the caching etc in place? Or will we have to basically re-do the FE code?

**Comment by radavis** [comment]
The front-end work is complete.

**Comment by wesrowe** [comment]
I suggest we close the ticket then – we have an epic in progress and tickets to get the back end in place. If the FE needs update later we'll identify that need.
